<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav)" />
  <arg name="profiler" default="$(optenv PROFILER false)" />
  <arg name="nodename" default="uvdar_driver" />

  <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- with empty serial number, the nodelet will list possibilities and shutdown -->
  <!-- <arg name="serial_number" default=""/> -->
  <arg name="frame_id" default="uvdar"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg UAV_NAME)_uvdar_manager" />
  <arg name="n_threads" default="8" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <group ns="$(arg UAV_NAME)">

    <!-- launch the nodelet -->
    <node pkg="nodelet" type="nodelet" name="$(arg nodename)" args="$(arg nodelet) uvdar_ros_driver/UvdarRosDriver $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <rosparam file="$(find uvdar_ros_driver)/config/config.yaml" />

      <param name="uav_name" type="string" value="$(arg UAV_NAME)"/>
      <param name="enable_profiler" type="bool" value="$(arg profiler)" />
      <param name="debug_serial" type="bool" value="false" />

      <param name="uvdar_usb_serial" type="string" value="$(arg serial_number)"/>
      <param name="uvdar_frame_id" type="string" value="$(arg frame_id)"/>

      <!-- remap the topic names to the UAV namespace
      <remap from="~distance" to="~received_message" />
      -->

    </node>

  </group>

</launch>
